posts/2020-05-15-five-years-of-rust.md

```diff
@@ -0,0 +1,508 @@
+To learn more, run the command again with --verbose.
+```
+
+</details>
+
+![A terminal screenshot of the 1.43.0 error message.][borrow-error-1.43.0]
+
+[borrow-error-1.2.0]:  /images/2020-05-15-five-years-of-rust/borrow-error-1.2.0.png
+[borrow-error-1.43.0]: /images/2020-05-15-five-years-of-rust/borrow-error-1.43.0.png
+[help-error-1.2.0]:    /images/2020-05-15-five-years-of-rust/help-error-1.2.0.png
+[help-error-1.43.0]:   /images/2020-05-15-five-years-of-rust/help-error-1.43.0.png
+[trait-error-1.2.0]:   /images/2020-05-15-five-years-of-rust/trait-error-1.2.0.png
+[trait-error-1.43.0]:  /images/2020-05-15-five-years-of-rust/trait-error-1.43.0.png
+
+## Quotes from the teams
+Of course we can't cover every change that has happened. So we reached out and
+asked some of our teams what changes they are most proud of:
+
+> For rustdoc, the big things were:
+> * The automatically generated documentation for blanket implementations
+> * The search itself and its optimizations (last one being to convert it into JSON)
+> * The possibility to test more accurately doc code blocks "compile_fail,
+>   should_panic, allow_fail"
+> * Doc tests are now generated as their own seperate binaries.
+>
+> — Guillaume Gomez ([rustdoc])
+
+
+> Rust now has baseline IDE support! Between IntelliJ Rust, RLS and
+> rust-analyzer, I feel that most users should be able to find "not horrible"
+> experience for their editor of choice. Five years ago, "writing Rust" meant
+> using old school Vim/Emacs setup.
+>
+> — Aleksey Kladov ([IDEs and editors][ides])
+
+
+> For me that would be: Adding first class support for popular embedded
+> architectures and achieving a striving ecosystem to make micro controller
+> development with Rust an easy and safe, yet fun experience.
+>
+> — Daniel Egger ([Embedded WG][ewg])
+
+
+> The release team has only been around since (roughly) early 2018, but even in
+> that time, we've landed ~40000 commits just in rust-lang/rust without any
+> significant regressions in stable.
+>
+> Considering how quickly we're improving the compiler and standard libraries, I
+> think that's really impressive (though of course the release team is not the
+> sole contributor here). Overall, I've found that the release team has done an
+> excellent job of managing to scale to the increasing traffic on issue
+> trackers, PRs being filed, etc.
+>
+> — Mark Rousskov ([Release][release])
+
+
+> Within the last 3 years we managed to turn Miri from an experimental
+> interpreter into a practical tool for exploring language design and finding
+> bugs in real code -- a great combination of PL theory and practice.  On the
+> theoretical side we have [Stacked Borrows], the most concrete proposal for a
+> Rust aliasing model so far.
+>
+> On the practical side, while initially we had to prompt maintainers to add
+> Miri checking to a few key libraries, recently we saw a great uptake of people
+> using Miri to [find and fix bugs] in various crates, and a similar uptake in
```

Comment:
```suggestion
> On the practical side, while initially only a few key libraries were checked in
> Miri by us, recently we saw a great uptake of people using Miri to
> [find and fix bugs] in their own crates and dependencies, and a similar uptake in
```
I'm worried otherwise it sounds too much like we were force-feeding people. ;)